src/supla/sensor/HX711.h

```diff
@@ -0,0 +1,161 @@
+/*
+ Copyright (C) AC SOFTWARE SP. Z O.O.
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ as published by the Free Software Foundation; either version 2
+ of the License, or (at your option) any later version.
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+*/
+
+#ifndef SRC_SUPLA_SENSOR_HX711_H_
+#define SRC_SUPLA_SENSOR_HX711_H_
+
+/*
+Dependency: https://github.com/olkal/HX711_ADC,
+use library manager to install it
+*/
+
+#include <HX711_ADC.h>
+
+#include "weight.h"
+#include "../storage/storage.h"
+
+namespace Supla {
+namespace Sensor {
+
+#pragma pack(push, 1)
+struct HX711ConfigData {
+  uint32_t tareOffset;
+  float calValue;
+};
+#pragma pack(pop)
+
+class HX711 : public Weight {
+ public:
+  HX711(int sdaPin, int sclPin, float defCalValue = 696.0)
+      : sdaPin(sdaPin), sclPin(sclPin), defCalValue(defCalValue),
+          hx711Sensor(sdaPin, sclPin) {
+  }
+
+  double getValue() {
+    if ((hx711Sensor.getSPS() < 7) || (hx711Sensor.getSPS() > 100)) {
+      value = WEIGHT_NOT_AVAILABLE;
+      SUPLA_LOG_DEBUG("HX711 sampling rate is abnormal, check connection");
+    } else {
+      value = (value < 0 && value > -1) ? 0 : value;
+    }
+    return value;
+  }
+
+  void onInit() {
+    hx711Sensor.begin();
+    hx711Sensor.setTareOffset(tareOffset);
+    uint64_t stabilizingTime = 2000;
+    bool _tare = false;
+    hx711Sensor.start(stabilizingTime, _tare);
+    if (hx711Sensor.getTareTimeoutFlag()) {
+      SUPLA_LOG_DEBUG("HX711 timeout, check connection");
+    } else {
+      calFactor = (calValue > 0) ? calValue : defCalValue;
+      hx711Sensor.setCalFactor(calFactor);
+      SUPLA_LOG_DEBUG("HX711 startup is complete");
+    }
+    while (!hx711Sensor.update()) {}
+    SUPLA_LOG_INFO("HX711 calibration value: %.2f", hx711Sensor.getCalFactor());
+    SUPLA_LOG_INFO(
+        "HX711 measured conversion time ms: %.2f",
+         hx711Sensor.getConversionTime());
+    SUPLA_LOG_DEBUG(
+        "HX711 measured sampling rate HZ: %.2f", hx711Sensor.getSPS());
+    SUPLA_LOG_INFO(
+        "HX711 measured settlingtime ms: %d", hx711Sensor.getSettlingTime());
+    if ((hx711Sensor.getSPS() < 7) || (hx711Sensor.getSPS() > 100)) {
+      SUPLA_LOG_DEBUG("HX711 sampling rate is abnormal, check connection");
+    }
```

Comment:
this check is not needed here. The same code will be executed in getValue which is called one line below

Reply:
corrected